tfmesos/scheduler.py

```diff
@@ -359,23 +372,55 @@ def registered(self, driver, framework_id, master_info):
             )
 
     def statusUpdate(self, driver, update):
-        mesos_task_id = int(update.task_id.value)
-        if update.state != 'TASK_RUNNING':
+        logger.debug('Received status update %s', str(update.state))
+        mesos_task_id = update.task_id.value
+        if self._is_terminal_state(update.state):
             task = self.tasks[mesos_task_id]
```

Comment:
Maybe we need to check if `mesos_task_id` is in `self.tasks`? Because update message may be late and the task has been already removed from `self.tasks`?